src/components/Product/Product.jsx

```diff
@@ -1,33 +1,169 @@
 import Image from 'next/image';
+import { useForm } from 'react-hook-form';
 import Typography from '../UI/Typography';
 import Container from '../UI/StyledContainer';
 import ImageWrapper from '../UI/StyledImageWrapper';
 import TextWrapper from '../UI/StyledTextWrapper';
 import DefaultButton from '../UI/StyledButton';
+import FormContainer from '../UI/Form/StyledFormContainer';
+import FormElement from '../UI/Form/StyledFormElement';
+import InputContainer from '../UI/Form/StyledInputContainer';
+import InputSingleContainer from '../UI/Form/StyledInputSingleContainer';
+import Label from '../UI/Form/StyledLable';
+import Input from '../UI/Form/StyledInput';
+import ButtonContainer from '../UI/StyledButtonContainer';
+import { useState } from 'react';
+import { useEffect } from 'react';
 
-export default function Product({ id, title, detail, email, image, altText, onDeleteProduct }) {
+export default function Product(props) {
+	const [isEditMode, setIsEditMode] = useState(false);
+
+	const enableEditMode = () => {
+		setIsEditMode(true);
+	};
+	const disableEditMode = () => {
+		setIsEditMode(false);
+	};
+
+	return (
+		<div>
+			{isEditMode ? (
+				<ProductModeEdit {...props} onDisableEditMode={disableEditMode} />
+			) : (
+				<ProductModeShow {...props} onEnableEditMode={enableEditMode} />
+			)}
+		</div>
+	);
+}
+
+function ProductModeShow({
+	id,
+	title,
+	detail,
+	email,
+	image,
+	altText,
+	onDeleteProduct,
+	onEnableEditMode,
+}) {
 	const handleDelete = () => {
 		onDeleteProduct(id);
 	};
-
 	return (
 		<Container>
 			<ImageWrapper>
-				<Image
-					src={image}
-					alt={altText}
-					width={3456}
-					height={5184}
-					layout="fill"
-					objectFit="cover"
-				/>
+				<Image src={image} alt={altText} layout="fill" objectFit="cover" />
 			</ImageWrapper>
 			<TextWrapper>
 				<Typography variant="p">{title}</Typography>
 				<Typography variant="p">{detail}</Typography>
-				<DefaultButton type="submit">{email}</DefaultButton>
+				<DefaultButton>{email}</DefaultButton>
 				<DefaultButton onClick={handleDelete}>delete</DefaultButton>
+				<DefaultButton onClick={onEnableEditMode}>edit</DefaultButton>
 			</TextWrapper>
 		</Container>
 	);
 }
+
+function ProductModeEdit({ id, title, detail, email, image, onDisableEditMode, onUpdateProduct }) {
+	const {
+		setValue,
+		watch,
+		register,
+		handleSubmit,
+		formState: { errors },
+	} = useForm();
+
+	const {
+		image: imageValue,
+		detail: detailValue,
+		email: emailValue,
+		title: titleValue,
+	} = watch();
+
```

Comment:
Is this `watch()` necessary?

Reply:
yes it's necessary, on line 153 you can see that I'm using the *Values to overwrite my props. And you need watch when using form hook to "watch your values". If you're not using form hook you can work with usestate and create a usestate for every single prop value